src/components/package/FileExplorer.vue

```diff
@@ -0,0 +1,359 @@
+          .filter(file => {
+            return (
+              file.fileLocation.replace(new RegExp("/$"), "") ===
+              this.filesLocation.replace(new RegExp("/$"), "")
+            );
+          })
+          .sort((a, b) => a.fileName.localeCompare(b.fileName));
+      },
+      //get folders in the current directory
+      currentDirectories () {
+        const dirs = [];
+
+        for (const file of this.fileSystem) {
+          const locationWithoutLastSlash = this.filesLocation.replace(
+            new RegExp("/$"),
+            "",
+            ),
+            dirToPush = file.fileLocation
+              .replace(locationWithoutLastSlash, "")
+              .split("/")[1];
+
+          if (
+            !dirs.includes(dirToPush) &&
+            file.fileLocation.includes(locationWithoutLastSlash) &&
+            dirToPush !== ""
+          ) {
+            dirs.push(dirToPush);
+          }
+        }
+        return dirs.sort((a, b) => a.localeCompare(b));
+      },
+      //get the parent dir
+      //we use this cuz utilizing "./" in the router-link does not work always, because is based on router history
+      parentDir () {
+        const routeWithoutSlashEnding = this.$route.path.endsWith("/")
+          ? this.$route.path.replace(new RegExp("/$"), "")
+          : this.$route.path;
+
+        return routeWithoutSlashEnding.substr(
+          0,
+          routeWithoutSlashEnding.lastIndexOf("/"),
+        );
+      },
+      //get the currently opened file's extension
+      currentFileExtension () {
+        const routeWithoutSlashEnding = this.$route.path.endsWith("/")
+          ? this.$route.path.replace(new RegExp("/$"), "")
+          : this.$route.path,
+          fileName = routeWithoutSlashEnding.split("/")[
+          routeWithoutSlashEnding.split("/").length - 1
+            ];
+
+        return fileName.split(".")[fileName.split(".").length - 1];
+      },
+      //get the number of lines of the currently opened file
+      //we use this for the line numbering with the code
+      fileContentLines () {
+        //when a json file is opened, axios auto-converts it into a javascript object
+        //we stringfiy the object to get the number of lines
+        //this doesn't work always, since many people have different styles when using json files
+        if (this.currentFileExtension === "json")
+          return JSON.stringify(this.currentFileContent, null, 4).split(
+            /\r\n|\r|\n/,
+          ).length;
+
+        return this.currentFileContent.split(/\r\n|\r|\n/).length;
+      },
+    },
+    filters: {
+      removeSlash (val) {
+        return val.replace(new RegExp("/", "g"), "");
+      },
+    },
+    async created () {
+      await this.reloadFiles();
+      await this.loadCurrentFile();
+      this.checkIfDirOrFileExists();
+    },
+    methods: {
+      //reload the files, and the filesystem on changes
+      async reloadFiles () {
+        console.log(this.name, this.version)
```

Comment:
Somewhy, these fields are undefined, if the files route is loaded first. The explorer still works, but it throws the silent error in the console.
This does not occur when loading the file explorer by clicking on "Browse Files" on the main page of the modules

Reply:
This must occur because when the created function is called, the props are still undefined